fastlane_bot/modes/base_triangle.py

```diff
@@ -9,301 +9,83 @@
-                            base_direction_one,
-                            x_match_curves_not_carbon,
-                            flt,
-                            arb_mode,
-                            combos,
-                        )
-                    if len(base_direction_two) > 0:
-                        combos = self.get_miniverse(
-                            y_match_curves_not_carbon,
-                            base_direction_two,
-                            x_match_curves_not_carbon,
-                            flt,
-                            arb_mode,
-                            combos,
-                        )
-        return combos
-    
-    def get_all_relevant_pairs_info(self, CCm, all_relevant_pairs):
-        # Get pair info for the cohort to allow decision making at the triangle level
-        all_relevant_pairs_info = {}
-        for pair in all_relevant_pairs:            
-            all_relevant_pairs_info[pair] = {}
-            pair_curves = CCm.bypair(pair)
-            carbon_curves = []
-            non_carbon_curves = []
-            for x in pair_curves:
-                if x.params.exchange in self.ConfigObj.CARBON_V1_FORKS:
-                    carbon_curves += [x]
-                else:
-                    non_carbon_curves += [x]
-            all_relevant_pairs_info[pair]['non_carbon_curves'] = non_carbon_curves
-            all_relevant_pairs_info[pair]['carbon_curves'] = carbon_curves
-            all_relevant_pairs_info[pair]['curves'] = non_carbon_curves + [carbon_curves] if len(carbon_curves) > 0 else non_carbon_curves  # condense carbon curves into a single list
-            all_relevant_pairs_info[pair]['all_counts'] = len(pair_curves)
-            all_relevant_pairs_info[pair]['carbon_counts'] = len(carbon_curves)
-        return all_relevant_pairs_info
-
-    def get_analysis_set_per_flt(self, flt, valid_triangles, all_relevant_pairs_info):
-        flt_triangle_analysis_set = []
-        for triangle in valid_triangles:
-            multiverse = [all_relevant_pairs_info[pair]['curves'] for pair in triangle]
-            product_of_triangle = list(itertools.product(multiverse[0], multiverse[1], multiverse[2]))
-            triangles_to_run = flatten_nested_items_in_list(product_of_triangle)
-            flt_triangle_analysis_set += list(zip([flt] * len(triangles_to_run), triangles_to_run))
-        
-        self.ConfigObj.logger.debug(f"[base_triangle.get_analysis_set_per_flt] Length of flt_triangle_analysis_set: {flt, len(flt_triangle_analysis_set)}")
-        return flt_triangle_analysis_set
-
-    def get_comprehensive_triangles(
-        self, flashloan_tokens: List[str], CCm: Any
-    ) -> Tuple[List[str], List[Any]]:
-        """
-        Get comprehensive combos for triangular arbitrage
+    @abc.abstractmethod
+    def get_combos(self, flashloan_tokens: List[str], CCm: Any, arb_mode: str) -> Tuple[List[str], List[Any]]:
+        ...
```

Comment:
doc string? generally in abstract methods it is good to not only define the signature but also the semantics.

Reply:
Abstract method removed